spec/filters/mutate_spec.rb

```diff
@@ -108,6 +108,71 @@
     end
   end
 
+  describe "select only one specified fields" do
+    config '
+      filter {
+        mutate {
+          select => [ "survivor" ]
+        }
+      }'
+
+    sample("remove-me" => "Goodbye!", "let-me-go" => "Cheers", "survivor" => "Hello.") do
+      insist { subject }.include?("survivor")
+      insist { subject }.include?("@timestamp")
+      reject { subject }.include?("remove-me")
+      reject { subject }.include?("let-me-go")
+    end
+  end
+
+  describe "select multiple specified fields" do
+    config '
+      filter {
+        mutate {
+          select => [ "survivor", "the-sequel" ]
+        }
+      }'
+
+    sample("remove-me" => "Goodbye!", "let-me-go" => "Cheers", "survivor" => "Hello", "the-sequel" => "World") do
+      insist { subject }.include?("survivor")
+      insist { subject }.include?("the-sequel")
+      insist { subject }.include?("@timestamp")
+      reject { subject }.include?("remove-me")
+      reject { subject }.include?("let-me-go")
+    end
+  end
+
+  describe "select should work on nested fields" do
+    config <<-CONFIG
+      filter {
+        mutate {
+          select => [ "[foo][bar]" ]
+        }
+      }
+    CONFIG
+
+    sample({ "foo" => { "bar" => "1000" }, "remove" => "me" }) do
+      insist { subject }.include?("[foo][bar]")
```

Comment:
Can we update `foo` to be `"foo" => { "bar" => "1000", "nomore" => "2000" }`

and then we can add an assertion like

```
reject { subject }.include?("[foo][nomore]")
```

something to that effect to explicitly reflect that other subfields of `foo` are not to be selected